NitroxModel/DataStructures/ThreadSafeQueue.cs

```diff
@@ -121,6 +121,22 @@ public IEnumerable<T> Clone()
             }
         }
 
+        public void Filter(Func<T, bool> predicate)
```

Comment:
Would rename to `RemoveWhere`. `Filter` is usually used when a method makes no side effect.